vortex-array/src/compute/conformance/binary_numeric.rs

```diff
@@ -41,44 +102,333 @@ where
+
+    // Test with special float values (NaN, Infinity)
+    test_binary_numeric_with_scalar(array.clone(), T::nan());
+    test_binary_numeric_with_scalar(array.clone(), T::infinity());
+    test_binary_numeric_with_scalar(array, T::neg_infinity());
+}
+
+fn test_binary_numeric_with_scalar<T>(array: ArrayRef, scalar_value: T)
+where
+    T: NativePType + Num + Copy + std::fmt::Debug,
+    Scalar: From<T>,
+{
+    let canonicalized_array = array
+        .to_primitive()
+        .vortex_expect("Failed to canonicalize array to primitive form for binary numeric test");
+    let original_values = to_vec_of_scalar(&canonicalized_array.into_array());
+
+    let scalar = Scalar::from(scalar_value)
+        .cast(array.dtype())
+        .vortex_unwrap();
+
+    // Only test operators that make sense for the given scalar
+    let operators = if scalar_value == T::zero() {
+        // Skip division by zero
+        vec![
+            NumericOperator::Add,
+            NumericOperator::Sub,
+            NumericOperator::RSub,
+            NumericOperator::Mul,
+        ]
+    } else {
+        vec![
+            NumericOperator::Add,
+            NumericOperator::Sub,
+            NumericOperator::RSub,
+            NumericOperator::Mul,
+            NumericOperator::Div,
+            NumericOperator::RDiv,
+        ]
+    };
+
+    for operator in operators {
+        // Test array operator scalar
+        let result = numeric(
+            &array,
+            &ConstantArray::new(scalar.clone(), array.len()).into_array(),
+            operator,
         );
+
+        // Skip if the entire operation fails
+        if result.is_err() {
+            continue;
+        }
+
+        let result = result.vortex_unwrap();
+        let actual_values = to_vec_of_scalar(&result);
+
+        // Check each element for overflow/underflow
+        let expected_results: Vec<Option<Scalar>> = original_values
+            .iter()
+            .map(|x| {
+                x.as_primitive()
+                    .checked_binary_numeric(&scalar.as_primitive(), operator)
+                    .map(<Scalar as From<PrimitiveScalar<'_>>>::from)
+            })
+            .collect();
+
+        // For elements that didn't overflow, check they match
+        for (idx, (actual, expected)) in actual_values.iter().zip(&expected_results).enumerate() {
+            if let Some(expected_value) = expected {
+                assert_eq!(
+                    actual,
+                    expected_value,
+                    "Binary numeric operation failed for encoding {} at index {} with scalar {:?}: \
+                     ({array:?})[{idx}] {operator:?} {scalar} \
+                     expected {expected_value:?}, got {actual:?}",
+                    array.encoding_id(),
+                    idx,
+                    scalar_value,
+                );
+            } else {
+                // For overflow elements, verify that the operation would indeed overflow
```

Comment:
same point here about overflow: I'm pretty sure we're re-running the command on the same value from original_values that caused us to take this branch.